pkg/types/spec/validations.go

```diff
@@ -1150,6 +1187,58 @@ func validateONNXPredictor(api *userconfig.API, models *[]CuratedModelResource,
 	return nil
 }
 
+func validateONNXModelFilePath(modelPath string, projectDir string, awsClient *aws.Client, gcpClient *gcp.Client) error {
+	s3Path := strings.HasPrefix(modelPath, "s3://")
+	gcsPath := strings.HasPrefix(modelPath, "gs://")
+	localPath := !s3Path && !gcsPath
+
+	if s3Path {
+		awsClientForBucket, err := aws.NewFromClientS3Path(modelPath, awsClient)
+		if err != nil {
+			return err
+		}
+
+		bucket, modelPrefix, err := aws.SplitS3Path(modelPath)
+		if err != nil {
+			return err
+		}
+
+		yes, err := awsClientForBucket.IsS3Prefix(bucket, modelPrefix)
```

Comment:
Should be checking if the prefix exists or if the file exists? E.g. if `modelPrefix` is 1/2, and there is no file named `1/2` but there is a file named `1/22` or `1/2/3` this will return true. If that is not desired, check out `IsS3File()`. If it is desired, disregard this comment :)

Reply:
The intention was to check the file, not the prefix. Fixed this! Thanks!